flink-core/src/main/java/org/apache/flink/util/ChildFirstClassLoader.java

```diff
@@ -73,9 +71,7 @@ public ChildFirstClassLoader(
 				// let URLClassLoader do it, which will eventually call the parent
 				c = super.loadClassWithoutExceptionHandling(name, resolve);
 			}
-		}
-
-		if (resolve) {
+		} else if (resolve) {
```

Comment:
what's this change about?

Reply:
It's a small performance optimization (should probably go in a separate hotfix commit). The typical class loading pattern indeed has this trailing resolution with an if, but in `ChildFirstClassLoader`, the resolution already happens at `super.loadClassWithoutExceptionHandling(name, resolve)`. So it should never be necessary to resolve it additionally.

Reply:
Ok that makes sense, let's move it into another commit and then we're good to merge.